File: app/src/main/java/me/imzack/app/end/injector/component/AllTypesComponent.java
package me.imzack.app.end.injector.component;

import me.imzack.app.end.injector.module.AllTypesPresenterModule;
import me.imzack.app.end.injector.scope.FragmentScope;
import me.imzack.app.end.view.fragment.AllTypesFragment;

import dagger.Component;

@FragmentScope
@Component(dependencies = AppComponent.class, modules = AllTypesPresenterModule.class)
public interface AllTypesComponent {

    void inject(AllTypesFragment allTypesFragment);
}

File: app/src/main/kotlin/me/imzack/app/end/injector/component/AllTypesComponent.kt
package me.imzack.app.end.injector.component

import dagger.Component
import me.imzack.app.end.injector.module.AllTypesPresenterModule
import me.imzack.app.end.injector.scope.FragmentScope
import me.imzack.app.end.view.fragment.AllTypesFragment

@FragmentScope
@Component(dependencies = arrayOf(AppComponent::class), modules = arrayOf(AllTypesPresenterModule::class))
interface AllTypesComponent {

    fun inject(allTypesFragment: AllTypesFragment)
}
